### PR TITLE
Use a RESTful API to retrieve job status

### PR DESCRIPTION
**Pull request by helicopter88**
And also, ensure that we do not print continuously "Executing KLEE" if the build is taking longer than one second

### CODE REVIEW

File: src/appjs/controllers.js [hunk 1]
```diff
@@ -99,6 +99,7 @@ controllers.controller('MainCtrl', [
             $scope.result = {};
             $scope.progress = [];
             $scope.progress.push('Job queued!');
+            var lastOutput = {};
```
Comment (andronat) [inline src/appjs/controllers.js]: You don't need that. (see next comment)

File: src/appjs/controllers.js [hunk 2]
```diff
@@ -109,12 +110,15 @@ controllers.controller('MainCtrl', [
...
                 channelId = data.taskId;
                 var fetch = $interval(function () {
-                  $http.get('/jobs/notify/?channel=' + channelId).success(function(data, status, headers) {
+                  $http.get('/jobs/status/' + channelId + '/').success(function(data, status, headers) {
                     m = angular.fromJson(data);
                     var type = m['type'];
                     if(type == 'notification') {
                       data = angular.fromJson(m.data);
-                      $scope.progress.push(data.message);
+                      if(data.message != lastOutput) {
```
Comment (andronat) [inline src/appjs/controllers.js]: Even if `data.message == lastOutput` then just update. no point on checking

File: src/klee_web/frontend/urls.py [hunk 3]
```diff
@@ -5,7 +5,8 @@
     url(r'^$', 'index', name='index'),
 
     # Web hooks
-    url(r'^jobs/notify/$', 'jobs_notify', name="jobs_notify"),
+    url(r'^jobs/notify/$', "jobs_notify", name="jobs_notify"),
```
Comment (andronat) [inline src/klee_web/frontend/urls.py]: If you want to replace `''` with `""` you need to make the change universal and be very careful as `''` and `""` are not exactly the same. I would prefer you do a separate PR with only styling change and not mixing functionality changes with style changes. Remove this change from this PR and as I said if you want to make style changes make a separate PR only for this reason through the whole code base.

File: src/klee_web/frontend/views.py [hunk 4]
```diff
@@ -44,7 +52,7 @@ def jobs_notify(request):
         task = Task.objects.get(task_id=channel)
         return HttpResponse(task.current_output)
     else:
-        return HttpResponseNotFound("This page only supports GET and POST")
+        return HttpResponseNotFound("This page only supports POST")
```
Comment (andronat) [inline src/klee_web/frontend/views.py]: If this method supports only `POST` then shouldn't you remove the `elif` in this [line](https://github.com/klee/klee-web/pull/79/files#diff-4855befe1940e201dbef5dae1d7cc435R50)?

**Comment by andronat** [comment]
CircleCI still fails with the same error which I thought it was fixed with this 1798e88e9843298439a7ab56a31930621ec1d8c9. Maybe it's not a versioning issue after all? :/

**Comment by andronat** [comment]
This PR refs: #77

**Comment by helicopter88** [comment]
Well, CircleCI passed once, and when given the same commit again, it failed, me and @giacomoguerci have tried hard this morning to get it to a deterministic state.
The elif with GET was deleted, probably brought it back by mistake when rebasing, will be removed.

**Comment by helicopter88** [comment]
Rebased, now passes everything

**Comment by andronat** [comment]
Just replace `channelId` with `jobId` and we can merge it

**Comment by helicopter88** [comment]
Done

**Comment by andronat** [comment]
@helicopter88 Don't forget to fix the issue on where we store the information.

**Pull request merged**